texar/torch/losses/info_loss.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2020 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Informational Theory Losses.
+"""
+import torch
+import torch.nn.functional as F
+
+
+__all__ = [
+    "kl_divg_loss_with_logits",
+]
+
+
+def kl_divg_loss_with_logits(
+    tgt_logits: torch.Tensor,
+    input_logits: torch.Tensor,
+    softmax_temperature: float = 1,
+    confidence_threshold: float = -1,
+    reduction: str = "mean",
+):
+    r"""
+    This function calculates the Kullback-Leibler divergence
+    between a pair of logits of the distributions.
+
+    It supports confidence-based masking and distribution sharpening.
+    Please refer to the UDA paper for more details.
+    (https://arxiv.org/abs/1904.12848)
+    Args:
+        - tgt_logits (Tensor):
+            A Tensor of arbitrary shape containing the target logits.
+        - input_logits (Tensor):
+            A Tensor of the same shape as tgt_logits
+            containing the input logits.
+        - softmax_temperature (float, optional):
+            The softmax temperature for sharpening the distribution.
+        - confidence_threshold (float, optional):
+            The threshold for confidence-masking. It is a threshold
+            of the probability in [0, 1], rather than of the logit.
+            If set to -1, the threshold will be ignored.
+        - reduction (optional):
+            This is the same as the `reduction` argument in :torch_docs:
+            `torch.nn.functional.kl_div <https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.kl_div>`  # pylint: disable=line-too-long
+            Specifies the reduction to apply to the output:
+            'none' | 'batchmean' | 'sum' | 'mean'.
+            'none': no reduction will be applied
```

Comment:
The list may not format well in the html documentation. Use the same format here: https://github.com/asyml/texar-pytorch/blob/master/texar/torch/core/layers.py#L585-L592 (i.e., a colon, a blank line, then a bullet list).